src/TypeInference/Pattern.mli

```diff
@@ -23,6 +23,12 @@ val check_type :
 val infer_arg_scheme :
   Env.t -> S.arg -> Env.t * T.pattern * T.scheme * ret_effect
 
+(** Infer type-scheme of given formal optional argument. Returns extended
+  environment, a pattern that represents an argument, its scheme, and the
+  effect of pattern-matching *)
+val infer_optional_arg_scheme :
+  Env.t -> S.arg -> Env.t * T.pattern * T.scheme * ret_effect
```

Comment:
Should this function be public? I didn't find any place where it was used outside this module.